project/cmake/scripts/windows/Macros.cmake

```diff
@@ -14,9 +15,15 @@ function(core_link_library lib wraplib)
     endforeach()
 
     add_library(wrap_${lib} SHARED ${SOURCES_ABS})
-    set_target_properties(wrap_${lib} PROPERTIES OUTPUT_NAME ${wraplib})
+    set_target_properties(wrap_${lib} PROPERTIES OUTPUT_NAME lib${lib})
     target_include_directories(wrap_${lib} PRIVATE ${INCLUDES})
     target_compile_definitions(wrap_${lib} PRIVATE ${DEFINITIONS})
+    
+    if(MSVC)
```

Comment:
Ah you're right I forget to test NMake.

If I understand [CMake's documentation](https://cmake.org/cmake/help/v3.0/variable/MSVC.html) `MSVC` is only set when you use the Visual Studio generator.